server/mergin/tests/test_middleware.py

```diff
@@ -0,0 +1,33 @@
+# Copyright (C) Lutra Consulting Limited
+#
+# SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-MerginMaps-Commercial
+
+import gevent
+import pytest
+
+from ..app import create_simple_app, GeventTimeoutMiddleware
+from ..config import Configuration
+
+
+@pytest.mark.parametrize("use_middleware", [True, False])
+def test_use_middleware(use_middleware):
+    """Test using middleware"""
+    Configuration.GEVENT_WORKER = use_middleware
+    Configuration.GEVENT_REQUEST_TIMEOUT = 1
+    application = create_simple_app()
+
+    def ping():
```

Comment:
A general point about testing (no need to modify anything here) but how sure are we that gevent has actually terminated the underlying request when it times out? It would be useful to create a test endpoint that (when active) generates high CPU load, perhaps with a tiny sleep so it's not completely blocking, and test that when a request to it times out, that CPU load does indeed drop back down.

Reply:
I did test with both, blocking and non-blocking functions generating high CPU load. In case of gevent timeout, load dropped after 504 return code. In case of blocking action, 200 was returned but whole process had to finish with high cpu load.

Reply:
Perfect!